src/Components/Bookings/BookingsModal/BookingsModal.js

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import "./BookingsModal.css";
+
+const BookingsModal = (props) => {
+	return (
+		<div key={props.bookingId} className="booking-modal">
+			<div>
+				<span>
+					<label>BookingID: {props.bookingId}</label>
+				</span>
+			</div>
+			<div>
+				<span>
+					<label>UserID: {props.userId}</label>
+				</span>
+			</div>
+			<div>
+				<span>
+					<label>Trip: {props.direction}</label>
+				</span>
+			</div>
+			<div>
+				<span>
+					<label>Date: {props.date}</label>
+				</span>
+			</div>
+			<span>
+				<label>Completed: {props.completed}</label>
+			</span>
+			<div className="booking-actions">
+				<button className="btn" onClick={props.onEdit.bind(props.bookingId)}>
+					Edit
+				</button>
+				<button
+					style={{ backgroundColor: "red" }}
+					className="btn"
+					onClick={props.onDelete.bind(props.bookingId)}>
```

Comment:
arrow functions don't have a `this` binding. Why are you binding?

Reply:
I tried to pass an argument/parameter through props but it didn't work. Binding worked for me.

Reply:
what do you mean by it didn't work? Why didn't it work? What error did you get? Did you google the error to find out why it didn't work?

Reply:
It's not that I'm getting an error message. From a tutorial I saw online, it was explained that the binding is used if you are passing a parameter to the function which would hold an argument when implemented in the parent element. Arrow function was also used. I have tried it without binding, the argument wasn't passed with the value (props.bookingId).

Reply:
You should be wary of tutorial and try to understand why they're doing that. In your own case, you need to learn **bind**, **apply**, and **call** in JavaScript especially as it relates to React.

In your own case, you want to pass the booking id for the delete and edit function. I think your approach works bu it's akin to Class-based approach. You can simplify this with **onDelete={() => props.onDelete(props.bookngId)}**